newIDE/app/src/VariablesList/index.js

```diff
@@ -48,6 +46,7 @@ type VariableAndName = {| name: string, ptr: number, variable: gdVariable |};
 
 type Props = {|
   variablesContainer: gdVariablesContainer,
+  inheritedVariablesContainer?: gdVariablesContainer,
```

Comment:
Almost good!
This should be: `inheritedVariablesContainer?: ?gdVariablesContainer`

This means the prop itself is optional (you're not forced to write `inheritedVariablesContainer={...}`) AND even if it is specified, you are allowed to pass something that is null (`inheritedVariablesContainer={null}` is authorized).
Here, you said `inheritedVariablesContainer?: gdVariablesContainer` which means: either you don't specify the prop or, if you specified it, then you are forced to pass me something that is not null.